sympy/printing/pycode.py

```diff
@@ -367,22 +367,6 @@ def _print_Stream(self, strm):
     def _print_NoneToken(self, arg):
         return 'None'
 
-
-class PythonCodePrinter(AbstractPythonCodePrinter):
-
-    def _print_sign(self, e):
-        return '(0.0 if {e} == 0 else {f}(1, {e}))'.format(
-            f=self._module_format('math.copysign'), e=self._print(e.args[0]))
-
-    def _print_Not(self, expr):
-        PREC = precedence(expr)
-        return self._operators['not'] + self.parenthesize(expr.args[0], PREC)
-
-    def _print_Indexed(self, expr):
-        base = expr.args[0]
-        index = expr.args[1:]
-        return "{}[{}]".format(str(base), ", ".join([self._print(ind) for ind in index]))
-
     def _hprint_Pow(self, expr, rational=False, sqrt='math.sqrt'):
```

Comment:
Can you explain why `_hprint_Pow` moved to the base class? That seems unrelated to the tests, as far as I can tell.

Reply:
In the original commit to this PR there were no class hierarchy changes or moving of methods, however, by adding a `_print_frac` method to the `PythonCodePrinter` (for the plain python / stdlib math.* implementation), I was required to add a `_print_frac`method to the `SymPyPrinter` (which at the time inherited from `PythonCodePrinter`); this was despite the fact that in master `SymPyPrinter` already handled `frac` adequately (printing it as `frac(x)` and not `x % 1` ) using its `_print_Function`. A better change than adding an override in `SymPyPrinter` (just to undo the newly introduced behaviour of its base class `PythonCodePrinter`) was to instead make `SymPyPrinter` inherit from `AbstractPythonCodePrinter` which bypasses the problem of inheriting incompatible printing intended for use with the stdlib math module. By making this change I then had to move `_hprint_Pow` to the base class so it can still be called by `SymPyPrinter`'s `_print_Pow`.

Reply:
Incidentally, another approach would be to put `_hprint_Pow` back in`PythonCodePrinter` and then simply deleting `SympyPrinter`'s `_print_Pow` method (but keeping the change to inherit from `AbstractPythonCodePrinter`). This will still allow the code `SymPyPrinter().doprint(sqrt(x))` to produce 'sqrt(x)'  but it will break one existing test `assert prntr._print_Pow(sqrt(x), rational=True) == 'x**(1/2)'`. I'm not sure this `rational=True` option is very useful for `SymPyPrinter` (which I believe is only used as a fallback printing method for lambdify when a sympy function has no printing implementation with e.g. scipy or mpmath) but for this PR at least keeping the existing test cases intact is probably best.

Reply:
Maybe printing with `sympy.sqrt` could have been introduced in my old PR #16901 in blind way. 
Sympy printer may not have to use sqrt and print everything as identical expressions. If the only part of test affected is `sympy.sqrt` it can be removed and _print_Pow can be removed too if this better simplifies the code.